RFCs/0000-expand-new-task.md

```diff
@@ -0,0 +1,164 @@
+# Expanded Bolt Task Authoring
+
+## Summary
+
+Implement various enhancements to the `pdk new task` command to make it easier to use PDK to implement Bolt Tasks that conform to current best practices.
+
+## Background & Assumptions
+
+- PDK has supported basic Bolt Task generation since version 1.2.0 which was released in October 2017.
+
+- Since that time, Bolt has continued to grow and evolve, and new helper libraries and best practices for writing Bolt Tasks have emerged.
+
+- See also: https://puppet.com/docs/bolt/latest/writing_tasks.html
+
+## Motivation & Goals
+
+PDK should provide up-to-date and robust tools for authoring all kinds of Puppet-related content, including Bolt Tasks. The current basic implementation of `pdk new task` can be improved in the following specific ways:
+
+- Users should be able to specify what language they plan to implement a new task in and PDK should generate a language-appropriate scaffold.
+
+- Users should be able to easily convert existing scripts into Bolt Tasks using PDK.
```

Comment:
convert or wrap?  I think it may be easier to automate copying the script to the files directory of a module with a small wrapper task that calls it.

Reply:
except for powershell which I think we have some hope of converting.

Reply:
Yeah I think given this and the other comment below I should clarify which convert/wrap workflows we are trying to support and clean things up.

Reply:
Deferring this motivation and the `--from` feature to a future RFC.